### PR TITLE
feat: Add body-parser(#39522)

### PR DESCRIPTION
**Pull request by SDmayank**

### CODE REVIEW

File: src/Server.ts [hunk 1]
```diff
@@ -1,28 +1,49 @@
...
+import { request } from 'http';
+
 class Server {
     app: express.Application;
-     constructor (private config: Iconfig) {
+    constructor(private config: Iconfig) {
         this.app = express();
     }
     bootstrap() {
+        console.log('Inside Bootstrap ');
```
Comment (VisaaliSuccessive) [inline src/Server.ts]: Remove unwanted console

File: src/Server.ts [hunk 2]
```diff
@@ -1,28 +1,49 @@
...
         this.setupRoutes();
         return this;
     }
-    setupRoutes(): void {
-     this.app.get('/`health-check', ( req: express.Request, res: express.Response) => {
-       res.send( 'Now app is running on the server' );
-    });
+    initBodyParser() {
+        const { app } = this;
+        console.log('inside the init bodyparser');
```
Comment (VisaaliSuccessive) [inline src/Server.ts]: Remove unwanted consoles

File: src/Server.ts [hunk 3]
```diff
@@ -1,28 +1,49 @@
...
+        this.app.get('/health-check', (req: express.Request, res: express.Response) => {
+            console.log('inside health check');
+            res.send('Now app is running on the server');
+        });
+        app.use('/body-parser', ( req: express.Request, res: express.Response) => {
+            console.log(req.body);
+            res.send('Your body parser is done');
+        });
+    app.use(notFoundRoute);
+    app.use(errorHandler);
```
Comment (VisaaliSuccessive) [inline src/Server.ts]: fix indentation issue here

File: src/libs/routes/errorHandler.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+const errorHandler = ((err , req , res , next) => {
+    console.log('Error', err);
+    res.send(
+        {
+            error : err.error,
+            code : err.code,
```
Comment (VisaaliSuccessive) [inline src/libs/routes/errorHandler.ts]: it must be status instead of code

File: src/libs/routes/errorHandler.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+const errorHandler = ((err , req , res , next) => {
+    console.log('Error', err);
+    res.send(
+        {
+            error : err.error,
```
Comment (VisaaliSuccessive) [inline src/libs/routes/errorHandler.ts]: there should be an optional error if you get any
 error: error || 'undefined'

File: src/libs/routes/errorHandler.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+            code : err.code,
+            message : err.error || err.message,
```
Comment (VisaaliSuccessive) [inline src/libs/routes/errorHandler.ts]: use destructuring for err

File: src/libs/routes/errorHandler.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+            timestamp : new Date(),
+        });
```
Comment (VisaaliSuccessive) [inline src/libs/routes/errorHandler.ts]: keep all these in a single variable and pass that in the response

File: src/libs/routes/errorHandler.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
+const errorHandler = ((err , req , res , next) => {
+    console.log('Error', err);
```
Comment (VisaaliSuccessive) [inline src/libs/routes/errorHandler.ts:2]: check whether headers are sent or not if not throw error

File: src/Server.ts [hunk 9]
```diff
@@ -1,28 +1,49 @@
 import * as express from 'express';
 import Iconfig from './config/Iconfig';
+import * as bodyParser from 'body-parser';
+import { errorHandler, notFoundRoute } from './libs/routes/index';
```
Comment (VisaaliSuccessive) [inline src/Server.ts]: no need to write index while importing. Remove it

File: src/Server.ts [hunk 10]
```diff
@@ -1,28 +1,49 @@
...
+    initBodyParser() {
+        const { app } = this;
+        console.log('inside the init bodyparser');
+        app.use(bodyParser.urlencoded({ extended: false }));
+        app.use(bodyParser.json());
     }
     run(): Server {
-        this.app.listen(this.config.port, ( err: any ): any => {
-            if ( err ) {
+        this.app.listen(this.config.port, (err: any): any => {
```
Comment (VisaaliSuccessive) [inline src/Server.ts:25]: use destructuring for this.config

**Review by VisaaliSuccessive — CHANGES_REQUESTED** [review]
Have you really run this code,  was this running properly? As I am unable to see any changes in your package.json file.

File: src/Server.ts (hunk 1 above)
Comment (SDmayank) [reply src/Server.ts]: done

File: src/Server.ts (hunk 2 above)
Comment (SDmayank) [reply src/Server.ts]: done

File: src/Server.ts (hunk 3 above)
Comment (SDmayank) [reply src/Server.ts]: done

File: src/libs/routes/errorHandler.ts (hunk 4 above)
Comment (SDmayank) [reply src/libs/routes/errorHandler.ts]: done

File: src/libs/routes/errorHandler.ts (hunk 5 above)
Comment (SDmayank) [reply src/libs/routes/errorHandler.ts]: done

File: src/libs/routes/errorHandler.ts (hunk 6 above)
Comment (SDmayank) [reply src/libs/routes/errorHandler.ts]: done

File: src/libs/routes/errorHandler.ts (hunk 7 above)
Comment (SDmayank) [reply src/libs/routes/errorHandler.ts]: done

File: src/libs/routes/errorHandler.ts (hunk 8 above)
Comment (SDmayank) [reply src/libs/routes/errorHandler.ts:2]: done

File: src/Server.ts (hunk 9 above)
Comment (SDmayank) [reply src/Server.ts]: done

File: src/Server.ts (hunk 10 above)
Comment (SDmayank) [reply src/Server.ts:25]: done

File: src/libs/routes/errorHandler.ts (hunk 7 above)
Comment (SDmayank) [reply src/libs/routes/errorHandler.ts]: > Have you really run this code, was this running properly? As I am unable to see any changes in your package.json file.

yes mam it was running properly

**Review by VisaaliSuccessive — APPROVED** [review]

**Pull request merged**